kubernetes/cluster_deploy.go

```diff
@@ -14,38 +15,77 @@ import (
 )
 
 func (c *Cluster) Deploy(options *ltops.DeployOptions) error {
-	log.Info("installing mattermost helm chart...")
-
-	if len(c.ReleaseName) > 0 {
-		log.Info("already installed as release '" + c.ReleaseName + "'")
-		return nil
+	previouslyDeployed := len(c.Release()) > 0
+	if previouslyDeployed {
+		log.Info("upgrading mattermost helm chart...")
+	} else {
+		log.Info("installing mattermost helm chart...")
 	}
 
 	license, err := ltops.GetFileOrURL(options.LicenseFile)
 	if err != nil {
 		return err
 	}
 
-	config, err := c.GetHelmConfigFromProfile(options.Profile, options.Users, string(license))
-	if err != nil {
-		return err
-	}
+	configFileLocation := ""
 
-	err = saveChartConfig(config, c.Config.WorkingDirectory)
-	if err != nil {
-		return err
+	if len(options.HelmConfigFile) == 0 {
+		config, err := c.GetHelmConfigFromProfile(options.Profile, options.Users, string(license))
+		if err != nil {
+			return err
+		}
+
+		err = saveChartConfig(config, c.Config.WorkingDirectory)
+		if err != nil {
+			return err
+		}
+
+		configFileLocation = filepath.Join(c.Config.WorkingDirectory, chartFilename)
+	} else {
+		configFileLocation = options.HelmConfigFile
 	}
 
-	cmd := exec.Command("helm", "install", "-f", filepath.Join(c.Config.WorkingDirectory, chartFilename), "mattermost/mattermost-helm")
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return errors.Wrap(err, "unable to install mattermost chart, error from helm: "+string(out))
+	if previouslyDeployed {
```

Comment:
What if it fails part way though? It would be nice if we didn't have to keep track of this and it could just work. Not sure fi that's a possibility.

Reply:
If it fails partway through the original deploy or the upgrade? As long as we have the release name the upgrade path will work

Reply:
Ok.